go-controller/pkg/node/helper_linux.go

```diff
@@ -61,6 +51,42 @@ func getDefaultGatewayInterfaceDetails() (string, []net.IP, error) {
 	return intfName, gatewayIPs, nil
 }
 
+func getDefaultGatewayInterfaceByFamily(family int) (string, net.IP, error) {
+	// filter the default route to obtain the gateway
+	filter := &netlink.Route{Dst: nil}
+	routes, err := netlink.RouteListFiltered(family, filter, netlink.RT_FILTER_DST)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "failed to get routing table in node")
+	}
+	// use the first valid default gateway
+	for _, r := range routes {
+		// no multipath
+		if len(r.MultiPath) == 0 {
+			intfLink, err := netlink.LinkByIndex(r.LinkIndex)
+			if err != nil {
+				klog.Warningf("Failed to get interface link for route %v : %v", r, err)
+				continue
+			}
+			klog.Infof("Found default gateway %s %s", intfLink.Attrs().Name, r.Gw.String())
```

Comment:
@aojea we should probably verify that `r.Gw != nil` too, since you don't need a specific gateway for a default route. If you don't, like with a point-to-point link or other odd scenarios, the other side of the link just takes that traffic and sends it along. It'll be pretty unlikely in a Kubernetes cluster, but maybe it'll happen. And in that case, would be nice to have a specific error message for it.

(also, previous code verified it was != nil, or at least skipped that route and tried subsequent ones)

Reply:
> If you don't, like with a point-to-point link or other odd scenarios, the other side of the link just takes that traffic and sends it along.

I will add the check, but just curious because I thought about this and could not understand how this can work without a gateway for a "default" route, 
So, you need to send packet to X, an address that you don´t have in your routing table, so you send it directly through an interface? what mac address destination you use to build the packet?